src/core/mount.c

```diff
@@ -66,6 +66,8 @@ static bool MOUNT_STATE_WITH_PROCESS(MountState state) {
                       MOUNT_CLEANING);
 }
 
+static Hashmap *mountinfo_hashmap;
```

Comment:
no. no global state in static variables like this, unless it's inherently a per-process thing, but this really isn't. 

global state should be placed in the Manager object, like everything else.

Reply:
> Anyway, this looks like optimization at the wrong place. If there's something slow with how mount units are processed here, and the logic does too much work then let's figure that out and suppress the work where it is done, but not just glue over it waaaaaay ahead of time by shortcutting everything.

I don't think the optimization is at the wrong place at all, it's necessary to have this because the notification we get from the kernel is not smart enough. Ideally if we could know which mounts changed or are new, then all this could be removed, but it's currently not the case.

Just have a look at the current code `device_found_node()` and `mount_setup_unit()`, you will see that it's very heavy work done there, reading udev database, etc for something we know (from the optimization) that it didn't change.

The systemtap statistics I provided speak by themselves. In the real world, the scenario wouldn't of course be a remount of a filesystem, but a new mount or unmount happening, which happens very often with Openshift or docker.